topics/RSA-Keys-Generation.md

```diff
@@ -0,0 +1,240 @@
+
+RSA-256, part of the RSA (Rivest–Shamir–Adleman) encryption algorithm, 
+utilizes a 256-bit key size to secure digital communications. 
+
+In the realm of [JSON Web Token](https://jwt.io/) authentication,
+RSA-256 plays a crucial role since the integrity and authenticity of JWTs can be verified through signature mechanisms, such as RSA-256, where a public/private key pair is employed.
+This ensures that the information contained within the token remains tamper-proof and trustworthy.
+
+In the following sections, we will cover how such keys are generated and used alongside the [JWT Ktor authentication](jwt.md) plugin.
+
+You can follow along by using the [auth-jwt-rs256](https://github.com/ktorio/ktor-documentation/blob/1.6.8/codeSnippets/snippets/auth-jwt-rs256/) sample project's private key,
+found in `resources/application.conf`. Simply save the private key in a `.pk8` file and [skip to the second](#second-step) step first step.
+
+## Generating an RSA private key
+
+To generate a private key, you can use `OpenSSL`, `ssh-keygen` or any other valid utility. For the rest of this page, we will
+focus on OpenSSL.
+
+<tabs>
+<tab title="OpenSSL">
+<code-block lang="shell">
+openssl genpkey -algorithm rsa &gt; ktor.pk8
+</code-block>
+</tab>
+</tabs>
+
+This command generates a private key using the RSA algorithm and stores it in the specified file, here `ktor.pk8`.
+The content of the file is [Base64](https://en.wikipedia.org/wiki/Base64) encoded and thus, needs to be
+decoded before we can derive its public key.
+
+## Deriving the public key {id="second-step"}
+
+In order to derive the public key from your previously generated private key, you will need to perform
+the following steps:
+
+1) Decode the private key
+2) Extract the public key
+3) Save the public key in PEM format
+
+This will give you the private key in `ktor.pk8`, and the public key in `ktor.spki`.
+
+Let's see how to do this using OpenSSL:
+
+<tabs>
+<tab title="OpenSSL">
+<code-block lang="shell">
+base64 -d &lt; ktor.pk8 | openssl pkey -inform der -pubout | tee ktor.spki
+</code-block>
+
+* `base64 -d < ktor.pk8`: This command decodes the Base64-encoded content of the `.pk8` file specified. The -d option is used for decoding.
+* `|`: Pipes the decoded output to the next command.
+* `openssl pkey -inform der -pubout`: This command reads the decoded content as [DER](https://en.wikipedia.org/wiki/X.690#DER_encoding) (Distinguished Encoding Rules) format, extracts the public key, and outputs it in [PEM](https://en.wikipedia.org/wiki/Privacy-Enhanced_Mail) (Privacy Enhanced Mail) format. The `-inform der` specifies the input format as DER, and `-pubout` indicates that the public key should be output.
+* `tee ktor.spki`: The `tee` command is used to redirect the output to both the console and a file named `ktor.spki`. This file will contain the public key in PEM format.
```

Comment:
Do you use the terminal output of `tee` somewhere? It seems you just use the file from now on, right?